File: Array | Dictionary | Set/contain_duplicate/js/contain_duplicate.js
console.log('Contain Duplicates')

/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const set = new Set();

    for ( num of nums ) {
        if ( ! set.has(num) ) {
            set.add(num)
        } else {
            return true;
        }
    }

    return false;
};

containsDuplicate([1,2,3,1]);

File: Array | Dictionary | Set/contain_duplicate/py/contain_duplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        dict = {}
        for num in nums:
            if not dict.get(num):
                dict[num] = 1
                # set.update({num:1})
            else:
                return True
        return False

sol = Solution()
print(sol.containsDuplicate([1,2,3,1]))
